packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -0,0 +1,581 @@
+// Minimum height of the segmented control.
+const double _kMinSegmentedControlHeight = 28.0;
+
+// Light partially-transparent blue color. Used to fill the background of a child option the user
+// is temporarily interacting with through a long press or drag.
+const Color _kPressedBackground = const Color(0x33007aff);
+
+/// An iOS-style segmented control.
+///
+/// Displays the widgets provided in the [LinkedHashMap] of [children] in a
+/// horizontal list. Used to select between a number of mutually exclusive
+/// options. When one option in the segmented control is selected, the other
+/// options in the segmented control cease to be selected.
+///
+/// A segmented control can feature any [Widget] as one of the values in its
+/// [LinkedHashMap] of [children]. The keys associated with these values are
+/// used to identify each widget and determine which widget is selected. As
+/// required by the [LinkedHashMap] class, keys must be of consistent types
+/// and must be comparable. The ordering of the keys will determine the order
+/// of the widgets in the segmented control.
+///
+/// When the state of the segmented control changes, the widget calls the
+/// [onValueChanged] callback. The key associated with the newly selected
+/// widget is returned in the [onValueChanged] callback. Most widgets that
+/// use a segmented control will listen for the [onValueChanged] callback
+/// and rebuild the segmented control with a new [groupValue] to update which
+/// option is currently selected.
+///
+/// The height of the segmented control is determined by the height of the
+/// tallest widget provided as a value in the [LinkedHashMap] of [children].
+/// The width of the segmented control is determined by the horizontal
+/// constraints on its parent. The available horizontal space is divided by
+/// the number of provided [children] to determine the width of each widget.
+/// The selection area for each of the widgets in the [LinkedHashMap] of
+/// [children] will then be expanded to fill the calculated space, so each
+/// widget will appear to have the same dimensions.
+///
+/// See also:
+///
+///  * <https://developer.apple.com/design/human-interface-guidelines/ios/controls/segmented-controls/>
+class SegmentedControl<T> extends StatefulWidget {
+  /// Creates an iOS-style segmented control bar.
+  ///
+  /// The [children] and [onValueChanged] arguments must not be null. Further,
+  /// the length of the [children] list must be greater than or equal to two.
+  ///
+  /// Each widget value in the map of [children] must have an associated key
+  /// that uniquely identifies this widget. This key is what will be returned
+  /// in the [onValueChanged] callback when a new value from the [children] map
+  /// is selected. Most widgets that use a segmented control will listen for
+  /// the [onValueChanged] callback and rebuild the segmented control with
+  /// the returned key as the new [groupValue] to update the currently selected
+  /// option.
```

Comment:
you can skip that third sentence since it doesn't really tell you about the constructor and it's redundant with what you said on the class anyway.